### PR TITLE
2ai documentation improvement

### PR DESCRIPTION
**Pull request by magicznyleszek**
This is first part of #580.

Changes:

- added `.editorconfig`
- added JSDoc config to for eslint (via plugin)
- moved most of `README.md` to separate documentation files
- new `npm run build-docs` command for building docs and is being run before every version published (via `publish-please`)
- adding _some_ JSDoc comments to files in repo
- fixing existing JSDoc comments that were invalid
- whitelisting module files with `package.json` `files`

After merging this, ~~we~~ you would need to enable GH page on __master branch /docs folder__ directory in repository's settings:

Screenshot 2019-04-13 at 20 26 26

Plus change the url in the repository's description.

### CODE REVIEW

**Comment by MartijnR** [comment]
Cool stuff! I was able to run it and view the generated docs. When you think you have a good basis (but of course with lots of missing/unclear documentation), I'd be open to just merge as phase I. We can hold off on linking from enketo.org until we think they are good enough.

Any thoughts on where we publish this? GitHub pages?

File: tutorials/index.json [hunk 1]
```diff
@@ -0,0 +1,26 @@
+{
+    "01-getting-started": {
+        "title": "Getting started"
+    },
+    "02-global-configuration": {
```
Comment (MartijnR) [inline tutorials/index.json]: let's call this "configuration"

File: tutorials/04-development.md [hunk 2]
```diff
@@ -0,0 +1,9 @@
+### How to develop Enketo Core
+
+1. install [node](http://nodejs.org/) version 8 until [this issue](https://github.com/enketo/enketo-transformer/issues/53) is closed, and [grunt-cli](http://gruntjs.com/getting-started)
+2. install dependencies with `npm install`
+3. build with `grunt`
+4. start built-in auto-reloading development server with `npm start`
+5. browse to [http://localhost:8005](http://localhost:8005/) and load an XForm url with the `xform` queryparameter or load a local from from the /tests/forms folder in this repo
+6. run tests with `npm test`
+7. adding the querystring `touch=true` and reducing the window size allows you to simulate mobile touchscreens
```
Comment (MartijnR) [inline tutorials/04-development.md]: ### Release a new version

Documentation is auto-generated and is re-built for each new release. Do not commit updated documentation in non-release commits. The process to follow for each release that includes various helpful checks is:

[ explain commands to use ]

[Explain the commands to use that perform the checks]

File: src/js/form.js [hunk 3]
```diff
@@ -828,7 +827,7 @@ Form.prototype.pathToAbsolute = function( targetPath, contextPath ) {
 
 /**
  * Validates question values.
- * 
+ *
  * @param  {jQuery} $input    [description]
  * @return {Promise}           [description]
```
Comment (MartijnR) [inline src/js/form.js]: This is just a question, but I'm wondering if JSDoc has a way to indicate what the data type of the final resolved value is.

File: README.md [hunk 4]
```diff
@@ -7,202 +7,6 @@ This repo is meant to use as a building block for any Enketo-powered application
 
 Follow the [Enketo blog](http://blog.enketo.org) or [Enketo on twitter](https://twitter.com/enketo) to stay up to date.
```
Comment (MartijnR) [inline README.md]: Add link to documentation

File: src/js/file-manager.js [hunk 5]
```diff
@@ -15,24 +15,24 @@ import { t } from 'enketo/translator';
```
Comment (MartijnR) [inline src/js/file-manager.js]: I think this module is not picked by the JSDoc

File: src/js/relevant.js [hunk 6]
```diff
@@ -89,7 +89,7 @@ export default {
             insideRepeat = clonedRepeatsPresent && $branchNode.parentsUntil( '.or', '.or-repeat' ).length > 0;
             insideRepeatClone = clonedRepeatsPresent && $branchNode.parentsUntil( '.or', '.or-repeat.clone' ).length > 0;
 
-            /*
```
Comment (MartijnR) [inline src/js/relevant.js]: I believe the relevant module is not picked by JSDoc.

**Comment by magicznyleszek** [comment]
@MartijnR sorry for delay in working on this 🙇 I made all the changes you requested plus added JSDoc comments to few more files that were missing from documentation. There is still some thing to do obviously, but please take a look if this is good to go as initial version of documentation :)

**Comment by MartijnR** [comment]
Yay! Thanks. Will check it out today!

**Comment by MartijnR** [comment]
It's live. Woohoo! 🎉  Will start going through code to change JSDoc comments.

**Pull request merged**